setup.py

```diff
@@ -0,0 +1,50 @@
+from setuptools import setup
+from setuptools_cpp import ExtensionBuilder, Pybind11Extension
+import os
+import glob
+
+# Currently a fake version for testing
+version = '0.0.1'
+
+# To enable consistency with sconsUtils we write the version to the
+# same location
+with open("./python/lsst/sphgeom/version.py", "w") as f:
+    print(f"""
+__all__ = ("__version__", )
+__version__ = '{version}'""", file=f)
+
+
+# read the contents of your README file
+this_directory = os.path.abspath(os.path.dirname(__file__))
+with open(os.path.join(this_directory, "README.md"), encoding='utf-8') as f:
+    long_description = f.read()
+
+# Find the source code
+pybind_src = sorted(glob.glob("python/lsst/sphgeom/*.cc"))
+cpp_src = sorted(glob.glob("src/*.cc"))
+
+
+def _get_module_name(path):
+    """Get the module part of the file name"""
+    file = os.path.split(path)[1]
+    root = os.path.splitext(file)[0]
+    if not root:
+        raise RuntimeError(f"Got {root} from {path} vis {file}")
+    return root
+
+
+# Very inefficient approach since this compiles the maing sphgeom
+# library code for every extension rather than building everything once
```

Comment:
FWIW, I don't think this is _as_ inefficient as it initially seems.  Or, rather, the best alternative isn't as much better as you might think (at least in general):

 - Concatenating all source into a single file, compiling a single object file, and linking that into a single .so might be the fastest approach for a package this small, because it means the compiler only processing the pybind11 headers once, but it doesn't scale to very large packages like afw, and it doesn't parallelize at all.

 - Compiling each source file into a separate object file, and then linking them into a single .so is what we recommend in new LSST code (and what we should switch to here, if not on this ticket); that enables much better warm rebuilds with scons (unchanged source files don't get recompiled) and much smaller total library sizes (-> better import times), but it does still require the compiler to process the entire set of pybind11 headers for every source file.

Reply:
I misunderstood some of this in the comments above.  Yes, recompiling all of sphgeom into each of these modules is definitely inefficient.

Reply:
Is this something that could be ameliorated by reviving the pybind11-fat-wrapper branch?

Reply:
@TallJimbo did mention that sphgeom isn't doing pybind11 how we are meant to do it these days. It would help if it was a single `_sphgeom.so` that was being imported it.  I might take a look at that since building and running all the tests in a minute is much better than 15 minutes.  I'm not entirely sure how to get setuptools to build the files in `src/` once and then use them (or how to get it to build a proper shared library and link against it).

Reply:
I'll see if I can rebase that branch.

Reply:
#26 has merged!

Reply:
Yay!